subprojects/gdk_rust/gdk_electrum/src/interface.rs

```diff
@@ -0,0 +1,946 @@
+        self.db.get_tip()
+    }
+
+    pub fn list_tx(&self, opt: &GetTransactionsOpt) -> Result<Vec<TransactionMeta>, Error> {
+        let (_, all_txs) = self.db.get_all_spent_and_txs()?;
+        let all_scripts = self.db.get_all_scripts()?;
+        let all_unblinded = self.db.get_all_unblinded()?; // empty map if not liquid
+
+        let mut txs = vec![];
+        let mut my_txids = self.db.get_my()?;
+        my_txids.sort_by(|a, b| b.1.unwrap_or(std::u32::MAX).cmp(&a.1.unwrap_or(std::u32::MAX)));
+
+        for (tx_id, height) in my_txids.iter().skip(opt.first).take(opt.count) {
+            debug!("tx_id {}", tx_id);
+
+            let tx = all_txs.get(tx_id).ok_or_else(fn_err(&format!("list_tx no tx {}", tx_id)))?;
+            let header = height
+                .map(|h| self.db.get_header(h)?.ok_or_else(fn_err("no header")))
+                .transpose()?;
+
+            let mut addressees = vec![];
+            for i in 0..tx.output_len() as u32 {
+                let script = tx.output_script(i);
+                if !script.is_empty() && !all_scripts.contains(&script) {
+                    let address = tx.output_address(i, self.network.id());
+                    addressees.push(AddressAmount {
+                        address: address.unwrap_or_else(|| "".to_string()),
+                        satoshi: 0, // apparently not needed in list_tx addressees
+                        asset_tag: None,
+                    });
+                }
+            }
+            let create_transaction = CreateTransaction {
+                addressees,
+                ..Default::default()
+            };
+
+            let fee = tx.fee(&all_txs, &all_unblinded);
+            let satoshi = tx.my_balances(&all_txs, &all_scripts, &all_unblinded);
+
+            let negatives = satoshi.iter().filter(|(_, v)| **v < 0).count();
+            let positives = satoshi.iter().filter(|(_, v)| **v > 0).count();
+            let (type_, user_signed) =
+                match (positives > negatives, tx.is_redeposit(&all_scripts, &all_txs)) {
+                    (_, true) => ("redeposit", true),
+                    (true, false) => ("incoming", false),
+                    (false, false) => ("outgoing", true),
+                };
+            let spv_verified = self.db.get_tx_verified(tx_id)?;
+
+            let tx_meta = TransactionMeta::new(
+                tx.clone(),
+                *height,
+                header.map(|h| h.time()),
+                satoshi,
+                fee,
+                self.network.id().get_bitcoin_network().unwrap_or(bitcoin::Network::Bitcoin),
+                type_.to_string(),
+                create_transaction,
+                user_signed,
+                spv_verified,
+            );
+
+            txs.push(tx_meta);
+        }
+        info!("list_tx {:?}", txs.iter().map(|e| &e.txid).collect::<Vec<&String>>());
+
+        Ok(txs)
+    }
+
+    fn utxos(&self) -> Result<WalletData, Error> {
```

Comment:
You are right, however, in the following MR gdk-electrum-2 it will contain only the utxo again, see 8b47af825c98bb900179364d3dbb9fca37b06b2f, so if you agree I will not change this here since it will be properly named at that point

Reply:
Agreed.